sql/src/main/java/io/crate/analyze/ConstraintsValidator.java

```diff
@@ -22,21 +22,50 @@
 
 package io.crate.analyze;
 
+import io.crate.core.collections.Maps;
 import io.crate.metadata.ColumnIdent;
 import io.crate.metadata.Reference;
 import io.crate.metadata.doc.DocTableInfo;
+import io.crate.types.DataTypes;
 
 import java.util.Collection;
 import java.util.Locale;
+import java.util.Map;
 
 public final class ConstraintsValidator {
 
-    public static void validate(Object value, Reference targetColumn) {
+    @SuppressWarnings("unchecked")
+    public static void validate(Object value, Reference targetColumn, DocTableInfo tableInfo) {
         assert targetColumn != null: "targetColumn is required to be able to validate it";
         // Validate NOT NULL constraint
         if (value == null && !targetColumn.isNullable()) {
             throw new IllegalArgumentException(String.format(Locale.ENGLISH,
-                "Cannot insert null value for column %s", targetColumn.ident().columnIdent().fqn()));
+                "Cannot insert null value for column '%s'", targetColumn.ident().columnIdent()));
+        }
+
+        // Validate NOT NULL constraint on object nested columns
+        if (targetColumn.valueType() == DataTypes.OBJECT) {
+            Map<String, Object> valueMap = (Map<String, Object>) value;
```

Comment:
could be extracted into a `validateNotNullOnChildren` method - would also remove the need for the comment